### PR TITLE
OAK-9721: Add state information while importing index

### PR DESCRIPTION
**Pull request by tihom88**
This is new PR for OAK-9721. You can find old PR at https://github.com/apache/jackrabbit-oak/pull/517/files
Created this PR as older code was in conflict with trunk's code.

### CODE REVIEW

File: oak-core/src/main/java/org/apache/jackrabbit/oak/plugins/index/importer/IndexImporter.java [hunk 1]
```diff
@@ -76,6 +78,10 @@
     private final IndexDefinitionUpdater indexDefinitionUpdater;
     private final boolean preserveCheckpoint = Boolean.getBoolean(OAK_INDEX_IMPORTER_PRESERVE_CHECKPOINT);
 
+    static final int RETRIES = Integer.parseInt(System.getProperty("oak.index.import.retries", "5"));
```
Comment (thomasmueller) [inline oak-core/src/main/java/org/apache/jackrabbit/oak/plugins/index/importer/IndexImporter.java]: FYI you could also use:

    Integer.getInteger("oak.index.import.retries", 5);

File: oak-core/src/main/java/org/apache/jackrabbit/oak/plugins/index/importer/IndexImporter.java [hunk 2]
```diff
@@ -90,72 +96,136 @@ public IndexImporter(NodeStore nodeStore, File indexDir, IndexEditorProvider ind
                 "checkpointed state [%s]", indexerInfo.checkpoint);
         this.indexDefinitionUpdater = new IndexDefinitionUpdater(new File(indexDir, INDEX_DEFINITIONS_JSON));
         this.asyncLaneToIndexMapping = mapIndexesToLanes(indexes);
+        this.indexPathsToUpdate = new HashSet<>();
+    }
+
+    enum IndexImportState {
+        NULL, SWITCHLANE, IMPORT_INDEX_DATA, BRING_INDEX_UPTODATE, RELEASE_CHECKPOINT
```
Comment (thomasmueller) [inline oak-core/src/main/java/org/apache/jackrabbit/oak/plugins/index/importer/IndexImporter.java]: Nit: SWITCH_LANE

**Review by thomasmueller — APPROVED** [review]

File: oak-core/src/main/java/org/apache/jackrabbit/oak/plugins/index/importer/IndexImporter.java (hunk 1 above)
Comment (tihom88) [reply oak-core/src/main/java/org/apache/jackrabbit/oak/plugins/index/importer/IndexImporter.java]: Done

File: oak-core/src/main/java/org/apache/jackrabbit/oak/plugins/index/importer/IndexImporter.java (hunk 2 above)
Comment (tihom88) [reply oak-core/src/main/java/org/apache/jackrabbit/oak/plugins/index/importer/IndexImporter.java]: Done

**Pull request merged**